extensions/cornerstone/src/getSopClassHandlerModule.js

```diff
@@ -102,13 +108,43 @@ function _getDisplaySetsFromSeries(instances, servicesManager, extensionManager)
     numInstances: 1,
     imageIdForThumbnail, // thumbnail image
     others: instances, // all other level instances in the image Pyramid
+    instances,
     othersFrameOfReferenceUID,
+    imageIds: instances.map(instance => instance.imageId),
   };
+  // The microscopy viewer directly accesses the metadata already loaded, and
+  // uses the DICOMweb client library directly for loading, so it has to be
+  // provided here.
+  const dicomWebClient = dataSource.retrieve.getWadoDicomWebClient?.();
+  const instanceMap = new Map();
+  instances.forEach(instance => instanceMap.set(instance.imageId, instance));
+  if (dicomWebClient) {
+    // This is really ugly - maybe this should be stored unmodified.
+    dicomWebClient.getDICOMwebMetadata = imageId => transferDenaturalizedDataset(denaturalizeDataset(fixMultivalueKeys(instanceMap.get(imageId))));
+
+    csUtils.genericMetadataProvider.add(displaySet.imageIds[0], {
+      type: 'webClient',
+      rawMetadata: dicomWebClient,
+    });
```

Comment:
This is a bit unusual, but I somewhat understand. Do you need to pass the DICOM web client along in the other places?

maybe have the type as `dicomWebClient` then, not webClient

Reply:
I think we need to stick to the convention and call the payload `metadata` as well